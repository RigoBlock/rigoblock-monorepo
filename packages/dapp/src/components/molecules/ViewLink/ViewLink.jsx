import './ViewLink.scss'
import Link, { LINK_SIZES } from '../../atoms/Link'
import PropTypes from 'prop-types'
import React from 'react'
import Tooltip, { ICON_SIZES } from '../../atoms/Icon'
import classNames from 'classnames'

const ViewLink = ({ icon, link, className }) => {
  const classProps = classNames('view-link', className)
  return (
    <Link size={LINK_SIZES.SMALL} to={link.to} className={classProps}>
      <Tooltip type={icon} size={ICON_SIZES.MEDIUM} />
      <span>{link.text}</span>
    </Link>
  )
}

ViewLink.propTypes = {
  icon: PropTypes.string.isRequired,
  link: PropTypes.object.isRequired,
  className: PropTypes.string
}

ViewLink.defaultProps = {
  className: ''
}

export default ViewLink
