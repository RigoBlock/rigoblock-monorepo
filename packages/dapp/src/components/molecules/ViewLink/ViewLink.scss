@import "../../_settings/variables.scss";

.view-link {
  display: flex;
  font-weight: $font-bold;
  line-height: $line-height;

  .tooltip {
    color: black;
  }

  &:visited {
    color: black;
  }

  &:active {
    color: red;
    .tooltip {
      color: red;
    }
  }

  :last-child {
    padding-left: $dist-xxs;
  }
}
