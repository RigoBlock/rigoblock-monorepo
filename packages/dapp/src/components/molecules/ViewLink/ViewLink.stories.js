import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import React from 'react'
import ViewLink from './ViewLink'

const props = {
  link: { text: 'Dashboard', to: '/' },
  icon: 'dashboard'
}

storiesOf('Molecules/ViewLink', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .add('default', () => <ViewLink {...props} />)
