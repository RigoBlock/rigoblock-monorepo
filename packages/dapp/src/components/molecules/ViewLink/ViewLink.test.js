import { shallow } from 'enzyme'
import ViewLink from './ViewLink.jsx'
import toJson from 'enzyme-to-json'

const props = { icon: 'dashboard', link: { text: 'Dashboard', to: '/' } }

describe('ViewLink component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(ViewLink, props)))
    ).toMatchSnapshot()
  })
})
