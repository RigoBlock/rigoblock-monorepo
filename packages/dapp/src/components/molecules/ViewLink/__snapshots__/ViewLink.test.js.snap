// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewLink component renders correctly 1`] = `
<Link
  className="view-link"
  size="small"
  to="/"
>
  <Tooltip
    onClick={[Function]}
    size="medium"
    tooltipText=""
    type="dashboard"
  />
  <span>
    Dashboard
  </span>
</Link>
`;
